packages/client/auth/src/CrossmintAuthClient.ts

```diff
@@ -0,0 +1,182 @@
+import type { UseSignInData } from "@farcaster/auth-kit";
+import {
+    type AuthMaterialWithUser,
+    CROSSMINT_API_VERSION,
+    CrossmintAuth,
+    type OAuthProvider,
+    REFRESH_TOKEN_PREFIX,
+    SESSION_PREFIX,
+} from "@crossmint/common-sdk-auth";
+import type { Crossmint } from "@crossmint/common-sdk-base";
+import { type CancellableTask, queueTask } from "@crossmint/client-sdk-base";
+import { deleteCookie, getCookie, getJWTExpiration, setCookie, TIME_BEFORE_EXPIRING_JWT_IN_SECONDS } from "./utils";
+
+export class CrossmintAuthClient extends CrossmintAuth {
+    private callbacks: CrossmintAuthClientCallbacks;
+    private refreshTask: CancellableTask | null = null;
+    private isRefreshing = false;
+
+    private constructor(crossmint: Crossmint, callbacks: CrossmintAuthClientCallbacks = {}) {
+        super(crossmint);
+        this.callbacks = callbacks;
+        this.handleRefreshToken();
```

Comment:
I would move this call to the `from` method, as this constructor is private and only called from there.
It's usually a bad practice to fire off processes from constructors, as the sole expected responsibility of them is to return an instance of the object.

Reply:
Fair, I explained in other comments why we're calling this

Reply:
Yes, thank you for that!
I do still believe it'd be cleaner for this call to be in the `from` method instead of the constructor. I do understand it has to be triggered on auth client initialization, what I suggest is merely moving it out of the constructor to keep the constructor clean of logic that doesn't exclusively represent constructing the instance.